iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -231,9 +243,9 @@ TEST_F(ServiceDiscovery_test, FindServiceWithInstanceAndEventWildcardReturnsAllM
     serviceContainerExp.push_back(SERVICE_DESCRIPTION3);
 
     auto serviceContainer = sut.findService(SERVICE, iox::capro::Wildcard, iox::capro::Wildcard);
-    ASSERT_FALSE(serviceContainer.has_error());
-    ASSERT_THAT(serviceContainer.value().size(), Eq(3U));
-    EXPECT_TRUE(serviceContainer.value() == serviceContainerExp);
+
+    ASSERT_THAT(serviceContainer.size(), Eq(3U));
+    EXPECT_TRUE(serviceContainer == serviceContainerExp);
```

Comment:
Those kind of tests are always shaky, since they rely on an implementation detail by assuming an order of returned services which could change. OK for now, but we should strive to write tests that do not use such kind of assumptions.

This could be fixed by e.g. sorting both containers before comparison.